src/Storages/StorageURL.cpp

```diff
@@ -624,7 +624,7 @@ void StorageURLSink::initBuffers()
     if (write_buf)
         return;
 
-    std::string content_type = FormatFactory::instance().getContentType(format, context, format_settings);
+    std::string content_type = FormatFactory::instance().getContentType(format, format_settings);
```

Comment:
Here we are trying to obtain the content type before initializing the format, which calls the offending function from FormatFactory.